opendbc/car/rivian/carcontroller.py

```diff
@@ -0,0 +1,40 @@
+from opendbc.can.packer import CANPacker
+from opendbc.car import Bus, apply_driver_steer_torque_limits, structs
+from opendbc.car.interfaces import CarControllerBase
+from opendbc.car.rivian.riviancan import create_lka_steering, create_longitudinal
+from opendbc.car.rivian.values import CarControllerParams
+
+LongCtrlState = structs.CarControl.Actuators.LongControlState
+
+class CarController(CarControllerBase):
+  def __init__(self, dbc_names, CP):
+    super().__init__(dbc_names, CP)
+    self.CP = CP
+    self.apply_steer_last = 0
+    self.packer = CANPacker(dbc_names[Bus.pt])
+
+  def update(self, CC, CS, now_nanos):
+    actuators = CC.actuators
+    can_sends = []
+
+    apply_steer = 0
+    if CC.latActive:
+      new_steer = int(round(CC.actuators.steer * CarControllerParams.STEER_MAX))
+      apply_steer = apply_driver_steer_torque_limits(new_steer, self.apply_steer_last,
+                                                     CS.out.steeringTorque, CarControllerParams)
+
+    # send steering command
+    self.apply_steer_last = apply_steer
+    can_sends.append(create_lka_steering(self.packer,  CS.acm_lka_hba_cmd, apply_steer, CC.latActive))
+
+    # Longitudinal control
+    if self.CP.openpilotLongitudinalControl:
+      stopping = actuators.longControlState == LongCtrlState.stopping
+      can_sends.append(create_longitudinal(self.packer, self.frame % 15, actuators.accel, CC.enabled, stopping))
```

Comment:
We can focus on longitudinal after the port is merged since you said you're still trying to get it to work